### PR TITLE
ui-kit v12

### LINKED ISSUE #1737: Upcoming breaking changes (v12)
## Summary

- [x] #1827 Drop Table
`<Table />` component has been deprecated in favor of `<DataTable />`. [reference](https://jira.commercetools.com/browse/UKIT-186)
- [x] #1828 Drop deprecated prop `constraint` on `Constraints.Horizontal` in favor of `max`. [reference](https://wiki.commercetools.com/display/shield/2020/11/17/New+Constraints+Scale)
- [x] #1825 Move RichTextEdit/LocalizedRichTextEditor out of beta [reference](https://jira.commercetools.com/browse/UKIT-194) 
- [x] #1824 Drop deprecated props on `Text` component
The `Text` component has a handful of deprecated props, such as `elementType` and `isInline`
- [x] #1826 Drop `hasUnderline` from `Link` (see #1548)
- [x] #1830 Drop deprecated prop `linkTo` in `SecondayButton`
  - remove `linkTo` from `<Tag />`

### PR DESCRIPTION
**Pull request by adnasa**
closes #1737

### CODE REVIEW

**Comment by adnasa** [comment]
The percy snapshots were approved for the constraint change and removal of `isInline` for `Text.Detail` 😎 
I don’t know if I missed something.

**Comment by adnasa** [comment]
### update

putting this on hold in favor of:

- address `Pagination` state bug
- introduce data-table-manager state hooks.

**Comment by adnasa** [comment]
@emmenko can I have a approval please 🤣

**Review by emmenko — APPROVED** [review]

**Pull request merged**
